pkg/kubelet/nodestatus/setters.go

```diff
@@ -73,16 +73,8 @@ func NodeAddress(nodeIP net.IP, // typically Kubelet.nodeIP
 			klog.V(2).Infof("Using node IP: %q", nodeIP.String())
 		}
 
-		if externalCloudProvider {
```

Comment:
since this code is already almost impossible to follow, minimal movement would be greatly appreciated. would this PR accomplish what it wants if it only made the following change? (if cloud is `nil` for external providers, I think it does)

```diff
- // We rely on the external cloud provider to supply the addresses.
- return nil
+ if len(node.Status.Addresses) > 0 {
+   // We rely on the external cloud provider to supply the addresses.
+   return nil
+ }
```

Reply:
Yup, can keep the changes minimal and accomplish the same thing. Will update